sys/include/xtimer.h

```diff
@@ -73,8 +73,10 @@ typedef void (*xtimer_callback_t)(void*);
  */
 typedef struct xtimer {
     struct xtimer *next;         /**< reference to next timer in timer lists */
+    uint32_t offset;             /**< lower 32bit offset time */
+    uint32_t long_offset;        /**< upper 32bit offset time */
+    uint32_t start_time;         /**< lower 32bit absolute start time */
     uint32_t target;             /**< lower 32bit absolute target time */
```

Comment:
Actually don't need a separate target variable if we allow slightly more computation to calculate "target" at each time it is needed.

Reply:
Ok then I'd favor removing the uint32_t and having the calculation instead, since compared to the rest of the calculation/update steps this calculation should be not very significant and the memory overhead is added for every timer alarm instance. Also deriving the value only from one space should be less error prone.

Reply:
Agree